### PR TITLE
Fix for NMDC request redirection

### LINKED ISSUE #96: DTS sends HTTP requests (not HTTPS) when running in a container in the Spin environment
Shreyas and I have been puzzling over some requests sent to NMDC's endpoints from the DTS Spin instance that are unencrypted HTTP requests, when they should be encrypted HTTPS requests. We've tried various things involving SSL certificates, but nothing has changed the situation.

I'm going to take some time over the winter break to audit the request code in the DTS. Specifically I intend to

- [x] set up a simple HTTPS client/server running locally and analyze network traffic between them
- [x] figure out how to configure the client to prevent the emission of unencrypted requests if it can't validate an SSL certificate on the server
- [x] replicate this setup with the client in a container and test until HTTPS requests work, building an understanding of how the container's configuration factors into this issue
- [x] update the DTS with safeguards to prevent any unencrypted HTTP requests from being generated
- [x] deploy the updated DTS in Spin and resume testing

Until then, the DTS Spin instance will not be running.

**Update: I've redeployed and restarted DTS in Spin with fixes from #98 .**

### PR DESCRIPTION
**Pull request by jeff-cohere**
This one's fun...

When you send a POST request to an NMDC endpoint with a trailing slash, it redirects HTTPS requests to HTTP, possibly causing your client to spray unencrypted data over the network! [Apparently this is an issue with FastAPI](https://github.com/fastapi/fastapi/discussions/9328). Meanwhile, [Django has problems if you POST without a trailing slash](https://dnmtechs.com/django-urls-without-trailing-slash-redirect-issue-in-python-3/). We're doomed. Dooomed, I say!

In this PR:

1. I tell the client to intercept redirects from HTTPS -> HTTP and emit errors.
2. I remove slashes from NMDC POST requests.

Onward...?

Closes #96

### CODE REVIEW

File: databases/nmdc/database.go [hunk 1]
```diff
@@ -284,9 +309,11 @@ type credential struct {
 }
 
 // fetches an access token / type from NMDC using a credential
-func getAccessToken(credential credential) (authorization, error) {
+func (db *Database) getAccessToken(credential credential) (authorization, error) {
 	var auth authorization
-	resource := baseApiURL + "token/"
+	// NOTE: no slash at the end of the resource, or there's an
+	// NOTE: HTTPS -> HTTP redirect (?!??!!)
```
Comment (ialarmedalien) [inline databases/nmdc/database.go:315]: You're gonna pay for forgetting that slash!!

**Comment by ialarmedalien** [comment]
Should there be some tests for this to check that the DTS does the right thing if the URL does/doesn't have a slash on it?

**Comment by jeff-cohere** [comment]
Fwiw, I just looked into creating a unit test for checking for a redirect, but it would involve breaking things apart in a way that only helps testing this one thing and complicates life for every other purpose. There are too many edge cases in web development!
